lib/cucumber/support_code/step_definition.js

```diff
@@ -106,19 +106,30 @@ function StepDefinition(pattern, options, code) {
 
       Cucumber.Util.Exception.registerUncaughtExceptionHandler(handleException, stepDomain);
 
+      var validCodeLengths = self.validCodeLengths(parameters);
+      if (validCodeLengths.indexOf(code.length) === -1) {
+        var error = 'definition has ' + code.length + ' arguments' +
```

Comment:
How about

"step definition has 3 arguments, should have 1 (if synchronous or returning a promise) or 2 (if accepting a callback)"?